### PR TITLE
nim doc now generates docs and runs runnableExamples for unexported symbols

### PR DESCRIPTION
**Pull request by nc-x**
Closes https://github.com/nim-lang/RFCs/issues/115

### CODE REVIEW

**Comment by Araq** [comment]
This should be done only for some new switch like ``--docPrivatSymbols``.

**Comment by nc-x** [comment]
Done

File: compiler/options.nim [hunk 1]
```diff
@@ -81,6 +81,7 @@ type                          # please make sure we have under 32 options
     optListFullPaths          # use full paths in toMsgFilename, toFilename
     optNoNimblePath
     optDynlibOverrideAll
+    optDocPvtSyms
```
[the remark this answers is not in the document]
Comment (nc-x) [reply compiler/options.nim]: Fixed.

**Comment by Araq** [comment]
Sorry I merged Zev's PR instead, wasn't aware we have two competing implementations. Please don't think yours was worse. :-)

**Comment by nc-x** [comment]
No problems.

**Pull request closed without merging**